### PR TITLE
feat: Mock Service Worker E2E Test Integration (#48)

### PR DESCRIPTION
**Pull request by Vroomfrondal**
- swapped out API calls for Mock Service Worker

### CODE REVIEW

**Comment by Vroomfrondal** [comment]
I think I did this correctly. The e2e tests are returning data I hardcoded in the handler get request instead of a live API call. And i'm getting the correct indications in the console that MSW is intercepting: 
Screenshot 2023-07-18 at 4 06 59 PM

I don't like how the `__mocks__` folder is in the `/src` directory when everything test related is in the `/cypress` folder. Was recommended in the docs but should I relocate it?

File: cypress/e2e/CRUD-Operations.cy.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
-// clear liked images on local storage
+import worker from '../../src/mocks/browser'
+
 before(() => {
   cy.clearAllCookies()
   cy.clearAllLocalStorage()
+
+  // Start MSW Mock Worker Handlers
+  worker.start({
```
Comment (RRamirez1414) [inline cypress/e2e/CRUD-Operations.cy.ts:8]: Smart to start the worker in your cypress test. If you were to write more cypress tests in the future, you could start the worker in the root react file `index.tsx` only if the application is running in a cypress environment. This way, you don't have to start it in every Cypress test. No need to implement it, just something you'll see in Addi

**Review by RRamirez1414 — APPROVED** [review]
Looks like you got the jist of it!  My knowledge is as good as yours if you were to move the msw worker files to the cypress folder. We have a test-utils folder in Addi also under src/ but now you got me curious. 
Im assuming as long as the worker is started in the cypress environment when the application is running, everything should be fine? Feel free to explore, but the initial setup looks good to me!

File: cypress/e2e/CRUD-Operations.cy.ts (hunk 1 above)
Comment (Vroomfrondal) [reply cypress/e2e/CRUD-Operations.cy.ts:8]: So this was the original method I had planned & also observed in Addi. However when using something like:
```ts
if (window.Cypress) { ... }
```

I was receiving the error `Property 'Cypress' does not exist on type 'Window & typeof globalThis'`. I wasn't sure if this was a typescript config issue or what. Did you guys do something special to access `Cypress` on the `window` object?

**Pull request merged**